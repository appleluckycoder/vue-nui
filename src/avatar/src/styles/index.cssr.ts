import { c, cE, cB } from '../../../_utils/cssr'

// vars:
// --n-font-size
// --n-border-radius
// --n-color
// --n-bezier
// --n-merged-size
export default cB('avatar', `
  width: var(--n-merged-size);
  height: var(--n-merged-size);
  color: #FFF;
  font-size: var(--n-font-size);
  display: inline-flex;
  position: relative;
  overflow: hidden;
  text-align: center;
  border-radius: var(--n-border-radius);
  background-color: var(--n-color);
  transition:
    background-color .3s var(--n-bezier),
    color .3s var(--n-bezier);
`, [
  c('img', `
    width: 100%;
    height: 100%;
  `),
  cE('text', `
    white-space: nowrap;
    display: inline-block;
    position: absolute;
    left: 50%;
    top: 50%;
  `),
  cB('icon', `
    vertical-align: bottom;
    font-size: calc(var(--n-merged-size) - 6px);
  `),
  cE('text', 'line-height: 1.25')
])
