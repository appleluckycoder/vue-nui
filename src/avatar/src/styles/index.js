import themedBaseStyle from './themed-base.cssr.js'
import themedSizeStyle from './themed-size.cssr.js'

export default [
  {
    key: 'mergedTheme',
    watch: [
      'mergedTheme'
    ],
    CNode: themedBaseStyle
  },
  {
    key: 'cssrSize',
    watch: [
      'mergedTheme',
      'cssrSize'
    ],
    CNode: themedSizeStyle
  }
]
