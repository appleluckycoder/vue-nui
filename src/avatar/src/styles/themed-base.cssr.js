import { c, cE, cTB } from '../../../_utils/cssr'

export default c([
  ({ props }) => {
    const {
      borderRadius,
      color: backgroundColor
    } = props.$local
    const {
      cubicBezierEaseInOut
    } = props.$base

    return cTB('avatar', {
      raw: `
        color: #FFF;
        font-size: 14px;
        display: inline-block;
        position: relative;
        overflow: hidden;
        text-align: center;
      `,
      borderRadius,
      transition: `
        background-color .3s ${cubicBezierEaseInOut},
        color .3s ${cubicBezierEaseInOut}
      `,
      backgroundColor
    }, [
      c('img', {
        width: '100%',
        height: '100%'
      }),
      cE('text', {
        raw: `
          white-space: nowrap;
          display: inline-block;
          position: absolute;
          left: 50%;
          top: 50%;
        `
      }),
      cTB('icon', {
        raw: `
          color: #FFF;
          vertical-align: bottom;
        `
      })
    ])
  }])
