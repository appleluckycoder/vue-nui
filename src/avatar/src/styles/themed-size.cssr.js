import { c, cE, cM, cTB, cB } from '../../../_utils/cssr'
import depx from '../../../_utils/css/depx'
import pxfy from '../../../_utils/css/pxfy'

export default c([
  ({ props }) => {
    const { size } = props.$instance
    const { height } = props.$local
    if (typeof size === 'number') {
      return null
    }
    const halfHeight = pxfy(depx(height[size]) / 2)

    return cTB('avatar', [
      cM(`${size}-size`, {
        height: height[size],
        width: height[size]
      }, [
        cM('circle-shaped', {
          borderRadius: halfHeight
        }),
        cM('round-shaped', {
          borderRadius: halfHeight
        }),
        cE('text', {
          lineHeight: 1.25
        }),
        cB('icon', {
          fontSize: height[size]
        })
      ])
    ])
  }
])
