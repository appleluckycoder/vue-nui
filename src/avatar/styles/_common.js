export default {
  heightTiny: '22px',
  heightSmall: '28px',
  heightMedium: '34px',
  heightLarge: '40px',
  heightHuge: '48px'
}
