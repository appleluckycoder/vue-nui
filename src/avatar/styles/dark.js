import create from '../../_styles/utils/create-component-base'
import commonVariables from './_common'

export default create({
  theme: 'dark',
  name: 'Avatar',
  getDerivedVariables ({ base, derived }) {
    const {
      borderRadius
    } = base
    const {
      avatarColorOverlay
    } = derived
    return {
      ...commonVariables,
      borderRadius,
      color: avatarColorOverlay
    }
  }
})
