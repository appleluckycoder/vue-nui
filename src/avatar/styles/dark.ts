import { commonDark } from '../../_styles/common'
import { AvatarTheme } from './light'

const avatarDark: AvatarTheme = {
  name: 'Avatar',
  common: commonDark,
  self (vars) {
    const {
      borderRadius,
      avatarColorOverlay,
      fontSize,
      heightTiny,
      heightSmall,
      heightMedium,
      heightLarge,
      heightHuge
    } = vars
    return {
      borderRadius,
      fontSize,
      heightTiny,
      heightSmall,
      heightMedium,
      heightLarge,
      heightHuge,
      color: avatarColorOverlay
    }
  }
}

export default avatarDark
