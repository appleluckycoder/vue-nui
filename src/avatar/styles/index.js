export { default as avatarDark } from './dark.js'
export { default as avatarLight } from './light.js'
