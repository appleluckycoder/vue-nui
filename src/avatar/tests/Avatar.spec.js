import { mount } from '@vue/test-utils'
import create from '../../create'
import { enUS } from '../../locales'
import { avatarLight } from '../styles'
import { NAvatar } from '../index'

describe('n-avatar', () => {
  const naive = create({
    locales: [
      enUS
    ],
    styles: [
      avatarLight
    ]
  })
  it('should work with import on demand', () => {
    mount(NAvatar, {
      global: {
        plugins: [naive]
      }
    })
  })
})
