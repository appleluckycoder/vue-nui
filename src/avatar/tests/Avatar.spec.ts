import { mount } from '@vue/test-utils'
import { NAvatar } from '../index'

describe('n-avatar', () => {
  it('should work with import on demand', () => {
    mount(NAvatar)
  })
})
