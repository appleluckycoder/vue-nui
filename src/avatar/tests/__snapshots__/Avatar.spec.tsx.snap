// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-avatar avatar adjust text 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\">adjust text</span></span>"`;

exports[`n-avatar custom style 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px; background-color: red;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\"></span></span>"`;

exports[`n-avatar icon avatar 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\"><i role=\\"img\\" class=\\"n-icon\\" style=\\"--bezier: cubic-bezier(.4, 0, .2, 1);\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><rect x=\\"32\\" y=\\"80\\" width=\\"448\\" height=\\"256\\" rx=\\"16\\" ry=\\"16\\" transform=\\"translate(512 416) rotate(180)\\" style=\\"fill: none; stroke: currentColor; stroke-linejoin: round; stroke-width: 32px;\\"></rect><line x1=\\"64\\" y1=\\"384\\" x2=\\"448\\" y2=\\"384\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></line><line x1=\\"96\\" y1=\\"432\\" x2=\\"416\\" y2=\\"432\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></line><circle cx=\\"256\\" cy=\\"208\\" r=\\"80\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></circle><path d=\\"M480,160a80,80,0,0,1-80-80\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></path><path d=\\"M32,160a80,80,0,0,0,80-80\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></path><path d=\\"M480,256a80,80,0,0,0-80,80\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></path><path d=\\"M32,256a80,80,0,0,1,80,80\\" style=\\"fill: none; stroke: currentColor; stroke-linecap: round; stroke-linejoin: round; stroke-width: 32px;\\"></path></svg></i></span></span>"`;

exports[`n-avatar image avatar 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px;\\"><img src=\\"https://07akioni.oss-cn-beijing.aliyuncs.com/07akioni.jpeg\\"></span>"`;

exports[`n-avatar round avatar 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 50%; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\"></span></span>"`;

exports[`n-avatar size is number 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 50px;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\"></span></span>"`;

exports[`n-avatar size is string 1`] = `"<span class=\\"n-avatar\\" style=\\"--font-size: 14px; --border-radius: 3px; --color: rgba(204, 204, 204, 1); --bezier: cubic-bezier(.4, 0, .2, 1); --size: 34px;\\"><span class=\\"n-avatar__text\\" style=\\"transform: translateX(-50%) translateY(-50%) scale(1);\\"></span></span>"`;
